src/core/thread/mle.cpp

```diff
@@ -247,6 +249,118 @@ ThreadError Mle::Stop(bool aClearNetworkDatasets)
     return kThreadError_None;
 }
 
+ThreadError Mle::Restore()
+{
+    ThreadError error = kThreadError_None;
+    NetworkInfo networkInfo;
+    uint16_t length;
+
+    SuccessOrExit(error = otPlatSettingsGet(mNetif.GetInstance(), kKeyNetworkInfo, 0,
+                                            reinterpret_cast<uint8_t *>(&networkInfo), &length));
+
+    VerifyOrExit(length == sizeof(networkInfo), error = kThreadError_NotFound);
+    VerifyOrExit(networkInfo.mDeviceState >= kDeviceStateChild, error = kThreadError_NotFound);
+
+    mDeviceMode = networkInfo.mDeviceMode;
+    mMac.SetChannel(networkInfo.mChannel);
+    mMac.SetPanId(networkInfo.mPanId);
+    mMac.SetExtendedPanId(networkInfo.mExtendedPanId.m8);
+    mMac.SetNetworkName(networkInfo.mNetworkName.m8);
+    SetMeshLocalPrefix(networkInfo.mMeshLocalPrefix.m8);
+    SetRloc16(networkInfo.mRloc16);
+    mMac.SetExtAddress(networkInfo.mExtAddress);
+    UpdateLinkLocalAddress();
+
+    mKeyManager.SetMasterKey(&networkInfo.mMasterKey, networkInfo.mMasterKeyLength);
+    mKeyManager.SetCurrentKeySequence(networkInfo.mKeySequence);
+
+    if (networkInfo.mDeviceState == kDeviceStateChild)
+    {
+        SuccessOrExit(error = otPlatSettingsGet(mNetif.GetInstance(), kKeyParentInfo, 0,
+                                                reinterpret_cast<uint8_t *>(&mParent), &length));
+    }
+
+#ifndef OPENTHREAD_MTD
+    else if (networkInfo.mDeviceState == kDeviceStateRouter || networkInfo.mDeviceState == kDeviceStateLeader)
+    {
+        otChildInfo childInfo;
+
+        mMleRouter.SetRouterId(GetRouterId(GetRloc16()));
+
+        for (uint8_t i = 0; i < kMaxChildren; i++)
+        {
+            SuccessOrExit(otPlatSettingsGet(mNetif.GetInstance(), kKeyChildInfo, i,
+                                            reinterpret_cast<uint8_t *>(&childInfo), &length));
+            SuccessOrExit(mMleRouter.AddChild(childInfo));
+        }
+    }
+
+#endif
+
+exit:
+    return error;
+}
+
+ThreadError Mle::Store()
+{
+    ThreadError error = kThreadError_None;
+    NetworkInfo networkInfo;
+    const uint8_t *key;
+
+    VerifyOrExit(IsAttached(), error = kThreadError_InvalidState);
+
+    memset(&networkInfo, 0, sizeof(networkInfo));
+
+    networkInfo.mDeviceMode = mDeviceMode;
+    networkInfo.mDeviceState = mDeviceState;
+
+    networkInfo.mChannel = mMac.GetChannel();
+    networkInfo.mPanId = mMac.GetPanId();
+    networkInfo.mRloc16 = GetRloc16();
+    memcpy(&networkInfo.mExtendedPanId, mMac.GetExtendedPanId(), sizeof(networkInfo.mExtendedPanId));
```

Comment:
I think we should use the stored Active Dataset for most of these values.  Of course, if there is no "official" Active Dataset, the device will need to create one with Active Timestamp set to 0.